rust-runtime/aws-smithy-xml/src/unescape.rs

```diff
@@ -47,33 +47,27 @@ pub fn unescape(s: &str) -> Result<Cow<str>, XmlError> {
                             // e.g. &#123;
                             (entity, 10)
                         } else {
-                            return Err(XmlError::InvalidEscape {
-                                esc: entity.to_string(),
-                            });
+                            return Err(XmlDecodeError::invalid_escape(entity));
                         };
                         let char_code = u32::from_str_radix(entity, radix).map_err(|_| {
-                            XmlError::InvalidEscape {
-                                esc: format!(
-                                    "Expected numeric escape in base {}; got: {}",
-                                    radix, &entity
-                                ),
-                            }
+                            XmlDecodeError::invalid_escape(format!(
+                                "expected numeric escape in base {}; got: {}",
+                                radix, &entity
+                            ))
```

Comment:
it'd be interesting if we could make these helper fns macros so that we could do:
```rust
XmlDecodeError::invalid_escape!("expected numeric escape in base {}; got: {}", radix, &entity)
```